### ISSUE TITLE
11.8.4 Reparaturassistenz - Anwendbarkeit

### ISSUE THREAD

**Issue by detlevhfischer** [BIK-BITV/BIK-Web-Test#150]
Der englische Text sagt:
> If the accessibility checking functionality of an authoring tool can detect that content does not meet a requirement of clauses 9 (...)  the authoring tool shall provide repair suggestion(s).

Wenn das Autorenwerkzeug keine solche Funktionalität besitzt, also nicht meckert, wenn jemand (über den Editor oder im Quelltext) eine Überschrift mit `style="font-size:40px; font-weight:bold"` umsetzt, dann ist der Prüfschritt nicht anwendar, richtig? Das müsste in "Anwendbarkeit" ergänzt werden.

Der Prüfschritt wäre also nur anwendbar, wenn es **überhaupt** "accessibility checking functionality" gibt, und würde sich dann mehr auf die Brauchbarkeit von Reparaturhinweisen beschränken?
Also: Es gibt eine Möglichkeit, ein Bild hochzuladen, und der Editor meckert beim Sichern "Bitte fügen sie im Feld "Alt" auch ein alt hinzu" (bewusst unklar ausgedrückt hier). Dann wäre die Anforderung, dass die Reparaturassistenz **hilfreich** sein sollte (also etwa "Bitte tragen sie im Feld "Alt" einen sinnvollen Alternativtext für Nutzer ein, die das Bild gegebenenfalls nicht sehen können (etwa blinde oder sehbehinderte Menschen)"). Gibt es gar keine Möglichkeit, für das Bild ein alt_Text zu definieren, wäre ja schon 11.8.2 nicht erfüllt).

**Comment by johannesFischer84** [comment]
> (Gibt es gar keine Möglichkeit, für das Bild einen Alternativ-Text zu definieren, wäre ja schon 11.8.2 nicht erfüllt).

Detlev, bei dem letzten Satz zu 11.8.2 stimme ich zu. Bezüglich 11.8.4 sehe ich es ein wenig anders. Dazu ist mir aufgefallen, dass die Formulierung bei den WCAG-Kriterien 3.3.1/3.3.3 (Fehlerbehandlung bei Eingabefehlern) leicht unterschiedlich zu EN 11.8.4 ist. In den WCAG heißt es:
> If an input error is automatically detected

In der EN heißt es:
> If the accessibility checking functionality of an authoring tool can detect that content does not meet

"**IS** detected" und "**CAN** detect" ist für mich ein wichtiger Unterschied. Die WCAG sagen Anwendbarkeit, wenn ein Fehler identifiziert wird, das heißt, es muss erst eine Fehlermeldung geben, dass Anwendbarkeit besteht. Die EN sagt durch das "can", dass die Fähigkeit bestehen muss, einen Fehler zu identifizieren. Die Fähigkeit und damit die Anwendbarkeit kann auch vorhanden sein, wenn das Werkzeug eigentlich gar keine Meldung ausgibt.

Bei der Nutzung ja/nein von Datenfeldern ist die Fähigkeit aus meiner Sicht auf jeden Fall erfüllt. Wenn also ein `alt` gesetzt oder nicht gesetzt wird bzw. ein Häkchen für dekorative Grafik, dann ist eine Software wohl immer leicht in der Lage dies abzuprüfen. Das Kriterium wäre dann anwendbar und eine Meldung muss bei Verstößen ausgegeben werden. Das die Meldung hilfreich sein muss, steht zwar nicht explizit in der EN, könnte man aber eventuell hineininterpretieren.

In deinem Beispiel der Gestaltung einer Überschrift über CSS mit `style="font-size:40px; font-weight:bold"` statt semantischer Auszeichnung wäre es schon schwieriger, den Fehler festzustellen. Zwar müsste aus meiner Sicht nach 11.8.2 die Möglichkeit gegeben sein, dass man bei gleichzeitiger Möglichkeit von Fett-Markierung und Einstellung der Schriftgröße auch eine Überschriften-Auszeichnung vornehmen kann. Aber die technische Fähigkeit, nach 11.8.4 einen Verstoß des Nutzers zu erkennen, ist aus meiner Sicht nur bedingt gegeben. Es gibt sicherlich Heuristiken wie die Länge des fetten, großen Texts im Vergleich zur Länge von übrigen Textabsätzen oder das Verhältnis der Schriftgröße von großem Text zu Schriftgröße von kleinem Text, aber wo soll da die klare Grenze gezogen werden? Eine Warnung des Werkzeugs ist sicherlich hilfreich und als Hinweis könnte man das sicherlich anmerken, vielleicht auch mit _eher erfüllt_ bewerten. Aber eine Nicht-Konformität würde ich wahrscheinlich nicht werten.

**Comment by MarcHaunschild** [comment]
Ich schließe mich @johannesFischer84 an - normalerweise kein Grund dafür auch einen Beitrag zu verfassen, aber ich möchte mich auch noch dafür bedanken. Ich wusste, ich hatte irgendwo gelesen, dass unter Umständen das fehlen von Fehlermeldungen ein fail ist, habe die Stelle aber nicht wiedergefunden. Jedenfalls habe ich auch unabhängig von Johannes das genau so verstanden wie er.

**Comment by sweckenmann** [comment]
> "Bitte fügen sie im Feld "Alt" auch ein alt hinzu" (bewusst unklar ausgedrückt hier). Dann wäre die Anforderung, dass die 

Reparaturassistenz hilfreich sein sollte (also etwa "Bitte tragen sie im Feld "Alt" einen sinnvollen Alternativtext für Nutzer ein, die das Bild gegebenenfalls nicht sehen können (etwa blinde oder sehbehinderte Menschen)"). 
Da in der Anforderung nur steht "shall provide repair suggestions" fände ich es ausreichend, wenn hier vermittelt wird "Bitte fügen sie im Feld "Alt" eine Textalternative für das Bild hinzu" - dein zusätzlicher Hinweis wäre besser, aber man kann jetzt nicht verlangen, dass bei jedem Hinweis noch eine Erklärung mitgegeben wird, warum das wichtig ist, ginge meiner Ansicht nach über die normative Anforderung hinaus.

> Wenn also ein alt gesetzt oder nicht gesetzt wird bzw. ein Häkchen für dekorative Grafik, dann ist eine Software wohl immer leicht in der Lage dies abzuprüfen.

Hm, das eine ist, ob es für eine Software grundsätzlich möglich ist, dies abzuprüfen oder ob die spezifische Software solche eine Abprüfung implementiert hat. "If an accessibility checking functionality..." - es könnte ja sein, es gibt keine solche Funktionalität? Würde man es nicht anders formulieren, wenn hier gefordert wird, dass automatisch prüfbare A11y Probleme eine Meldung erzeugen sollen?

**Comment by johannesFischer84** [comment]
> Hm, das eine ist, ob es für eine Software grundsätzlich möglich ist, dies abzuprüfen oder ob die spezifische Software solche eine Abprüfung implementiert hat. "If an accessibility checking functionality..." - es könnte ja sein, es gibt keine solche Funktionalität? Würde man es nicht anders formulieren, wenn hier gefordert wird, dass automatisch prüfbare A11y Probleme eine Meldung erzeugen sollen?

Auf Englisch würde ich noch die Formulierung "is able to detect" passend für die Fähigkeit finden. Aber bei der Formulierung "can detect" fällt mir außer der Fähigkeit (in der Lage sein) als andere übersetzte Bedeutung nur noch die Erlaubnis (dürfen) ein. Und diese Formulierung ("wenn die Funktionalität erkennen darf, dass ...") würde für mich keinen Sinn machen. Oder habe ich etwas übersehen?

**Comment by detlevhfischer** [comment]
@johannesFischer84 schrieb:
> Die EN sagt durch das "can", dass die Fähigkeit bestehen muss, einen Fehler zu identifizieren. Die Fähigkeit und damit die Anwendbarkeit kann auch vorhanden sein, wenn das Werkzeug eigentlich gar keine Meldung ausgibt.

Die EN sagt "If it can... then do X". Es gibt hier also im "If", im "Wenn..." die Bedingung, dass eine "accessibility checking functionality" des Tools vorhanden ist. Wenn so eine Funktionalität überhaupt nicht implementiert wurde (und das ist für uns im Einzelnen wohl nicht immer leicht feststellbar), sehe ich nicht, dass der Prüfschritt anwendbar wäre (oder wir gehen bewusst deutlich über die EN hinaus). 
Was den Punkt verstärkt: in der Anforderung geht es ja eher ums Inhaltliche, also um die Ausgabe von Reparaturhinweisen (was m.M. nach deutlich über den bloßen Einsatz etwa eines required-Attribut hinausgeht und damit eher an 3.3.3 Error Suggestion als an 3.3.1 Error detection erinnert) - und es wird ja gerade **nicht** das grundsätzliche Vorhandensein einer "accessibility checking functionality" verlangt?

**Comment by johannesFischer84** [comment]
> Es gibt hier also im "If", im "Wenn..." die Bedingung, dass eine "accessibility checking functionality" des Tools vorhanden ist. Wenn so eine Funktionalität überhaupt nicht implementiert wurde

Achso, jetzt verstehe ich, was ihr meint. Ich hatte den Worten "accessibility checking functionality" wohl nicht genügend Aufmerksamkeit geschenkt bzw. war ich davon ausgegangen, dass an sich die (Basis-)Funktionalität des Autorenwerkzeugs gemeint ist. Allerdings würde sie dann wohl nicht "accessibility checking functionality" genannt, sondern "authoring tool functionality".

Detlev, ich stimme dir damit im Großen und Ganzen zu, vielen Dank für das Stoßen auf den richtigen Punkt :-)
Inwieweit es mit 3.3.1 oder 3.3.3 der WCAG übereinstimmt, bin ich aber immer noch nicht sicher. Ich würde beides irgendwie einbeziehen, so explizit steht es für mich nicht in der Anforderung. Also eine Identifizierung des fehlerhaften Felds bzw. der fehlerhaften Eingabe muss schon vorhanden sein und der Anwender muss auch erfahren, wie er den Fehler korrigieren kann. Aber einen Hinweis, warum die Anforderung wichtig ist ("sinnvollen Alternativtext für Nutzer ein, die das Bild gegebenenfalls nicht sehen können (etwa blinde oder sehbehinderte Menschen)"), würde ich nicht verlangen. Das fordern nicht mal die WCAG bei Formularfehlern.

**Comment by MarcHaunschild** [comment]
Das war wirklich sehr interessant zu lesen. Ich habe mich raus gehalten, obwohl mir relativ schnell bei den Entgegnungen klar war, dass ich mich vermutlich geirrt hatte. Ich wollte aber nicht noch einmal Verwirrung stiften und habe mich daher zurück gehalten mit einer neuerlichen Zustimmung zu was auch immer ;-)
